framework/src/base/Attributes.C

```diff
@@ -309,21 +309,38 @@ AttribPreIC::isEqual(const Attribute & other) const
   return isMatch(other);
 }
 
-void
-AttribPreAux::initFrom(const MooseObject * /*obj*/)
-{
-}
 bool
 AttribPreAux::isMatch(const Attribute & other) const
 {
   auto a = dynamic_cast<const AttribPreAux *>(&other);
-  return a && (a->_val == _val);
+  if (!a || a->_vals.size() < 1)
```

Comment:
Why should this return false if _vals.size is zero?  If _vals.size() and other._vals.size() are both zero - then shouldn't it return true?  Or if you expect _vals to never be size zero than you should assert that here.

Reply:
Firstly, I'll note this new implementation of `AttribPreAux` is based entirely on the existing attribute `AttribExecOns`. They are now very similar classes.

If _vals.size() is zero, this function returns false because it means that there are no exec flags in `a` where `AttribPreAux` should return true so this is a short cut if statement as is present in `AttribExecOns::isMatch`.

Would it always be a programming error for `a` to have a _vals vector with `size()!=1`? I'm not sure about that. If it is an error, perhaps we should change 'AttribExecOns::isMatch' as well

Reply:
I think both being zero length either needs to be asserted against or resolve to returning true here.